File: restler-integration-tests/src/main/kotlin/org/restler/integration/springdata/PersonsRepository.java
package org.restler.integration.springdata;

import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.Param;
import org.springframework.data.rest.core.annotation.RepositoryRestResource;

import java.util.List;

@RepositoryRestResource(exported = true, path = "persons")
public interface PersonsRepository extends CrudRepository<Person, Long> {
    Person findById(@Param("id") Long id);

    List<Person> findByName(@Param("name") String name);

}

File: restler-integration-tests/src/main/kotlin/org/restler/integration/springdata/PersonsRepository.kt
package org.restler.integration.springdata

import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.query.Param
import org.springframework.data.rest.core.annotation.RepositoryRestResource
import java.lang

// java.lang.String - it is workaround for https://youtrack.jetbrains.com/issue/KT-5821

@RepositoryRestResource(exported = true, path = "persons")
interface PersonsRepository : CrudRepository<Person, lang.String> {

    fun findByName(@Param("name") name: String): List<Person>
    fun findById(@Param("id") id: lang.String): Person
}
